plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -0,0 +1,376 @@
+import static io.trino.plugin.jdbc.StandardColumnMappings.tinyintWriteFunction;
+import static io.trino.plugin.jdbc.StandardColumnMappings.varbinaryWriteFunction;
+import static io.trino.plugin.jdbc.StandardColumnMappings.varcharColumnMapping;
+import static io.trino.plugin.jdbc.StandardColumnMappings.varcharWriteFunction;
+import static io.trino.plugin.jdbc.TypeHandlingJdbcSessionProperties.getUnsupportedTypeHandling;
+import static io.trino.plugin.jdbc.UnsupportedTypeHandling.CONVERT_TO_VARCHAR;
+import static io.trino.spi.StandardErrorCode.NOT_FOUND;
+import static io.trino.spi.StandardErrorCode.NOT_SUPPORTED;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.BooleanType.BOOLEAN;
+import static io.trino.spi.type.DateType.DATE;
+import static io.trino.spi.type.DecimalType.createDecimalType;
+import static io.trino.spi.type.DoubleType.DOUBLE;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.RealType.REAL;
+import static io.trino.spi.type.SmallintType.SMALLINT;
+import static io.trino.spi.type.TimestampType.TIMESTAMP_MILLIS;
+import static io.trino.spi.type.TinyintType.TINYINT;
+import static io.trino.spi.type.VarcharType.createUnboundedVarcharType;
+import static java.lang.Integer.min;
+import static java.lang.String.format;
+import static java.lang.String.join;
+import static java.util.Locale.ENGLISH;
+
+public class ClickHouseClient
+        extends BaseJdbcClient
+{
+    public static final String ESCAPE_CHARACTER = "\"";
+    private static final Logger log = Logger.get(ClickHouseClient.class);
+
+    @Inject
+    public ClickHouseClient(BaseJdbcConfig config, ConnectionFactory connectionFactory, TypeManager typeManager)
+    {
+        super(config, ESCAPE_CHARACTER, connectionFactory);
+    }
+
+    @Override
+    public String quoted(RemoteTableName remoteTableName)
+    {
+        return quoted(
+                null,
+                remoteTableName.getSchemaName().orElse(null),
+                remoteTableName.getTableName());
+    }
+
+    @Override
+    protected String quoted(@Nullable String catalog, @Nullable String schema, String table)
+    {
+        StringBuilder sb = new StringBuilder();
+        if (!isNullOrEmpty(schema)) {
+            sb.append(quoted(schema)).append(".");
+        }
+        sb.append(quoted(table));
+        return sb.toString();
+    }
+
+    @Override
+    protected void copyTableSchema(Connection connection, String catalogName, String schemaName, String tableName, String newTableName, List<String> columnNames)
+    {
+        String sql = format(
+                "CREATE TABLE %s AS %s ",
+                quoted(catalogName, schemaName, newTableName),
+                quoted(catalogName, schemaName, tableName));
+        execute(connection, sql);
+    }
+
+    @Override
+    public JdbcOutputTableHandle createTable(ConnectorSession session, ConnectorTableMetadata tableMetadata, String tableName)
```

Comment:
We don't have to override the entire `createTable` for passing table properties, we just need to override - `createTableSql` where we could give a direct query String. Please take a look at `SqlServerClient` for reference.

Reply:
The argument `columns` of `createTableSql` does not include column is nullable or not ?
In ClickHouse, if a column is nullable, it must be specified with `columnName Nullable(colType)` 
such as `name Nullable(String)`.
So I have to override BaseJdbcClient#createTable method

Reply:
I guess if we need to provide custom column details, we can override `BaseJdbcClient#getColumnDefinitionSql` where we could get the `ColumnMetadata` and we could do similar logic. In that case we might not need to override `BaseJdbcClient#createTable`

Reply:
OK, I will try

Reply:
I have overrided `BaseJdbcClient#getColumnDefinitionSql` and remove overridden `BaseJdbcClient#createTable`